ddp-workspace/projects/ddp-testboston/src/app/components/enrollment/enrollment.component.ts

```diff
@@ -85,4 +87,50 @@ export class EnrollmentComponent implements OnInit {
     profile.lastName = this.accountForm.controls.lastName.value;
     return profile;
   }
+
+  private createStudyParticipant(invitationId: string): Observable<string> {
+    return this.subjectInvitation.createStudyParticipant(invitationId).pipe(
+      tap(userGuid => this.session.setParticipant(userGuid))
+    );
+  }
+
+  private isProfileChanged(profile: UserProfile): boolean {
+    return this.profile.firstName !== profile.firstName || this.profile.lastName !== profile.lastName;
+  }
+
+  private setupInitialState(): void {
+    const invitationId$ = of(this.session.session.invitationId);
+
+    invitationId$.pipe(
+      take(1),
+      mergeMap(invitationId => {
```

Comment:
Could this be simplified, dropping the creation of invitationId$ and just create the pipeline from `this.subjectInvitation(invitationId)`?

Reply:
Hm, no, I don't think so: I did it in this way, because CRCs can open testboston.org/enrolement directly, without selecting a study subject. So the component will try to get `invitationId` from the session if `invitationId` exists, will send a `lookupInvitation` request, otherwise, observable will throw an error to prevent the next requests and show the error message.